discovery-provider/alembic/versions/5bcbe23f6c70_user_track_collection_mat_views.py

```diff
@@ -0,0 +1,238 @@
+"""user-track-collection-mat-views
+
+Revision ID: 5bcbe23f6c70
+Revises: 2ff46a8686fa
+Create Date: 2021-04-12 20:01:40.395480
+
+"""
+from alembic import op
+import sqlalchemy as sa
+
+
+# revision identifiers, used by Alembic.
+revision = '5bcbe23f6c70'
+down_revision = '2ff46a8686fa'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    connection = op.get_bind()
+    connection.execute('''
+      begin;
+        --- ======================= AGGREGATE USER =======================
+        DROP MATERIALIZED VIEW IF EXISTS aggregate_user;
+        DROP INDEX IF EXISTS aggregate_user_idx;
+
+        CREATE MATERIALIZED VIEW aggregate_user as
+        SELECT
+            distinct(u.user_id),
+            COALESCE (user_track.track_count, 0) as track_count,
+            COALESCE (user_playlist.playlist_count, 0) as playlist_count,
+            COALESCE (user_album.album_count, 0) as album_count,
+            COALESCE (user_follower.follower_count, 0) as follower_count,
+            COALESCE (user_followee.followee_count, 0) as following_count,
+            COALESCE (user_repost.repost_count, 0) as repost_count,
+            COALESCE (user_track_save.save_count, 0) as track_save_count
+        FROM 
+            users u
+        -- join on subquery for tracks created
+        LEFT OUTER JOIN (
+            SELECT
+                t.owner_id as owner_id,
+                count(t.owner_id) as track_count
+            FROM
+                tracks t
+            WHERE
+                t.is_current is True AND
+                t.is_delete is False AND
+                t.is_unlisted is False AND
+                t.stem_of is Null
+            GROUP BY t.owner_id
+        ) as user_track ON user_track.owner_id = u.user_id
+        -- join on subquery for playlists created
+        LEFT OUTER JOIN (
+            SELECT
+                p.playlist_owner_id as owner_id,
+                count(p.playlist_owner_id) as playlist_count
+            FROM
+                playlists p
+            WHERE
+                p.is_album is False AND
+                p.is_current is True AND
+                p.is_delete is False AND
+                p.is_private is False
+            GROUP BY p.playlist_owner_id
+        ) as user_playlist ON user_playlist.owner_id = u.user_id
+        -- join on subquery for albums created
+        LEFT OUTER JOIN (
+            SELECT
+                p.playlist_owner_id as owner_id,
+                count(p.playlist_owner_id) as album_count
+            FROM
+                playlists p
+            WHERE
+                p.is_album is True AND
+                p.is_current is True AND
+                p.is_delete is False AND
+                p.is_private is False
```

Comment:
we wouldn't use this view for a user's own session?

Reply:
public album/playlist count shouldn't include private. technically private album isn't even a thing, but i think this is right